sympy/matrices/common.py

```diff
@@ -2377,23 +2393,27 @@ def __pow__(self, exp):
 
         return self.pow(exp)
 
-    def pow(self, exp, jordan=None, dotprodsimp=None):
+
+    def pow(self, exp, method="multiply"):
```

Comment:
Currently the default is to multiply except in the 2x2 case when the exponent is a large explicit integer. I think we should still have default=None for that case. The idea is that passing `method='multiply'` is a way of explicitly saying that you want to use the recursion method (even if it is a 2x2 with large exponent).

The None case can use heuristics to decide on the best method. We might find here that cayley is better for large explicit integer exponents.

Am I right in thinking that the cayley method doesn't handle symbols as exponents? E.g jordan can do:
```julia
In [5]: Matrix([[1, 2], [2, 1]]) ** n                                                                                                          
Out[5]: 
⎡     n    n         n    n⎤
⎢ (-1)    3      (-1)    3 ⎥
⎢ ───── + ──   - ───── + ──⎥
⎢   2     2        2     2 ⎥
⎢                          ⎥
⎢      n    n       n    n ⎥
⎢  (-1)    3    (-1)    3  ⎥
⎢- ───── + ──   ───── + ── ⎥
⎣    2     2      2     2  ⎦
```

Reply:
yes, cayley  doesnot support symbolic exp

Reply:
I think that the multiply method  should be removed. When the exp is number then based on the method `jordon` or `cayley` otherwise recursion should be used.

Reply:
When I described the multiply method I intended to mean that it is recursion. I just don't think that recursion is a good name to give to users.

Reply:
> jordon or cayley

The name is "Jordan" with an 'a'.